src/course/domain/course-id.ts

```diff
@@ -0,0 +1,15 @@
+export class CourseId {
+  constructor(private readonly value: string) {
+    if (!value || value.trim() === '') {
+      throw new Error('Course ID cannot be empty');
+    }
+    const uuidRegex =
+      /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/i;
```

Comment:
Do you think this is necessary? I think all this regular expresion is not necessary